elasticsearch/helpers/actions.py

```diff
@@ -659,18 +659,33 @@ def reindex(
     target_client = client if target_client is None else target_client
     docs = scan(client, query=query, index=source_index, scroll=scroll, **scan_kwargs)
 
-    def _change_doc_index(hits, index):
+    def _change_doc_index(hits, index, is_data_stream):
         for h in hits:
             h["_index"] = index
+            if is_data_stream:
+                h["_op_type"] = "create"
             if "fields" in h:
                 h.update(h.pop("fields"))
             yield h
 
     kwargs = {"stats_only": True}
     kwargs.update(bulk_kwargs)
+
+    is_data_stream = False
```

Comment:
Instead of auto-detecting if `target_index` is a data_stream, instead we can provide a new parameter `op_type` where if it's set will set `_op_type` to that value and then document that `op_type="create"` is needed for Data Streams.

What do you think of this approach?